src/base/TypeTraits.h

```diff
@@ -33,9 +33,18 @@ class Interface
     Interface(Interface &&) = default;
 };
 
```

Comment:
I have already done the equivalent for move operators before this PR was submitted. The PR is not performing any behaviour changes. The classes removing only copy are omitted here intentionally to avoid behaviour change.